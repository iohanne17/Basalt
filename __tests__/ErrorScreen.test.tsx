import React from 'react';
import ErrorScreen from '../src/lib/components/error';
import renderer from 'react-test-renderer';
import {render} from '@testing-library/react-native';

const props = {
  title: 'Network error',
  message: 'Try again later',
  onPress: () => null,
  isLoading: false,
};

test('Renders snapshot as expected', () => {
  const tree = renderer.create(<ErrorScreen {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render the label', () => {
  const {getByText} = render(<ErrorScreen {...props} />);

  const label = getByText('Network error');
  expect(label).toBeTruthy();
});
