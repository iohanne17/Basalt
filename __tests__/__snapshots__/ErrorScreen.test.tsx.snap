// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders snapshot as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FE812A",
        "fontFamily": "SchibstedGrotesk-Black",
        "fontSize": 18,
        "letterSpacing": -0.22,
        "lineHeight": 20,
      }
    }
    title={true}
  >
    Network error
  </Text>
  <Text
    style={
      Object {
        "color": "#FE812A",
        "fontFamily": "SchibstedGrotesk-Medium",
        "fontSize": 17,
        "letterSpacing": -0.22,
        "lineHeight": 24,
      }
    }
  >
    Try again later
  </Text>
  <View
    style={
      Array [
        Object {},
        Object {
          "height": 20,
          "width": undefined,
        },
      ]
    }
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FE812A",
          "borderRadius": 25,
          "justifyContent": "center",
        },
        Object {
          "opacity": 1,
        },
        Object {
          "height": 50,
          "width": 150,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "SchibstedGrotesk-Black",
          "fontSize": 18,
          "letterSpacing": -0.22,
          "lineHeight": 20,
        }
      }
      title={true}
    >
      Retry
    </Text>
  </View>
</View>
`;
